lib/Doctrine/DBAL/Configuration.php

```diff
@@ -39,12 +40,10 @@ public function setSQLLogger(SQLLogger $logger = null)
 
     /**
      * Gets the SQL logger that is used.
-     *
-     * @return \Doctrine\DBAL\Logging\SQLLogger|null
      */
-    public function getSQLLogger()
+    public function getSQLLogger() : SQLLogger
     {
-        return $this->_attributes['sqlLogger'] ?? null;
+        return $this->_attributes['sqlLogger'] ?? $this->_attributes['sqlLogger'] = new NullLogger();
```

Comment:
This needs to be documented too, apart from dropping nullability.

Reply:
Also setLogger should no longer be nullable or default to null.

Reply:
Actually calling `setSQLLogger(null)` really is a shortcut for `setSQLLogger(new NullLogger)` now, a quick way to remove the logger. I don't mind not allowing null, just wondering if it can be useful to anyone?

Reply:
Yes I understand, but it doesn't make much sense. The logger is no longer nullable, you can't _remove it_.

Reply:
If anything, defaulting to null is nonsense, calling `$configuration->setSQLLogger()` is just bad design.

Reply:
Sure, at least it would need to be `?SQLLogger`. But I'm OK to disallow `null` altogether, that's done. I also updated `UPGRADE.md`, and added `void` return type to `setSQLLogger()`.